### ISSUE TITLE
Booleans logical OR on data source merge

### ISSUE THREAD

**Issue by Steve-Louie-Bose** [hairyhenderson/gomplate#531]
Booleans do not follow precedence as specified in [datasource merge](https://docs.gomplate.ca/datasources/#using-merge-datasources). Instead they prefer true by computing a logical `or`.

### default.yaml
```
bool: true
convBool: "True"
```

### override.yaml
```
bool: false
convBool: "False"
```

### template.yaml
```
{{- $ex := (ds "example") }}
bool: {{ $ex.bool }}
convBool: {{ conv.ToBool $ex.convBool }}
```

`gomplate -d "example=merge:file://<PATH>/override.yaml|file://<PATH>/default.yaml" -f template.yaml -o out.yaml`

### out.yaml
```
bool: true
convBool: false
```

Even though the override specifies `false` the resulting output shows `true`. The mergo package states this is desired behavior due to go zero-values. [Issue](https://github.com/imdario/mergo/issues/24) 

## Resolution?
- Do not use bools, but use strings and [conv.ToBool](https://docs.gomplate.ca/functions/conv/#conv-tobool) as shown in the example.
- Maybe [WithOverride](https://godoc.org/github.com/imdario/mergo#WithOverride) can help? I believe it will reverse the precedence order currently in place on `merge`

**Comment by hairyhenderson** [comment]
Thanks for logging this @Steve-Louie-Bose! And thanks for the reference to the `mergo` issue.

Just to set expectations, I'm _not sure_ if this will be solvable very quickly. But my personal feeling is that this is a bug and that the value from the override datasource should always override the default, regardless of zero value.

I think this also means that other zero values won't override (like `""` or `0`).

**Comment by Steve-Louie-Bose** [comment]
No worries @hairyhenderson! I do believe a similar problem would arise for all zero values. 

`WithOverride` sounds promising in regards to a fix. I'm not certain of what ripple effects it would bring to the merge behavior.

**Comment by hairyhenderson** [comment]
Hmm... According to the docs I'm not sure `WithOverride` will work in this case:

> WithOverride will make merge override non-empty dst attributes with non-empty src attributes values.

We want to override with _empty_ attributes. But, I'll try this out!

**Comment by Steve-Louie-Bose** [comment]
> We want to override with empty attributes. But, I'll try this out!

My $0.02 is to not override empty. I'd set sane defaults in a template, and if necessary provide modifications or additions by merging files. To override on _empty_ would mean I have to provide all values again even if I want to keep them the same?

**Comment by hairyhenderson** [comment]
Oh, I think I misunderstood 🤦‍♂️... I was reading `empty` as `zero` (obviously not the same thing!) 😂

I'll play with this when I get a chance and see what I can figure out 😉

**Comment by hairyhenderson** [comment]
Quick update... `WithOverride` doesn't really work as expected - in fact, it seems to just invert the order (possible bug in mergo?).

I think I have a good solution though - I'm going to ~steal~borrow the merge code from Helm, which works as desired 😉

**Comment by hairyhenderson** [comment]
this is fixed in gomplate 3.4.0 now 🙂
